### ISSUE TITLE
VictoriaLogs `_time:` doesn't honor exclusive/inclusive range bounds `)` and `]`

### ISSUE THREAD

**Issue by eduardbareev** [VictoriaMetrics/VictoriaMetrics#6293]
### Describe the bug

```
curl --json '
{ "_msg": "m1", "_time":"2024-05-12T00:00:00.111+00:00" }
{ "_msg": "m2", "_time":"2024-05-12T00:00:00.222+00:00" }
{ "_msg": "m3", "_time":"2024-05-12T00:00:00.333+00:00" }
{ "_msg": "m4", "_time":"2024-05-12T00:00:00.444+00:00" }
{ "_msg": "m5", "_time":"2024-05-12T00:00:00.555+00:00" }
' 'http://localhost:9428/insert/jsonline'

sleep 3

# example 1

# requested range has inclusive upper bound, 
# but boundary point m3 is missing in result
curl 'http://localhost:9428/select/logsql/query' --data-urlencode \
    'query=_time:[0,2024-05-12T00:00:00.333+00:00]  '
{"_time":"2024-05-12T00:00:00.111Z","_stream":"{}","_msg":"m1"}
{"_time":"2024-05-12T00:00:00.222Z","_stream":"{}","_msg":"m2"}

# works as expected here
curl 'http://localhost:9428/select/logsql/query' --data-urlencode \
    'query=_time:[0,2024-05-12T00:00:00.333+00:00)  '
{"_time":"2024-05-12T00:00:00.111Z","_stream":"{}","_msg":"m1"}
{"_time":"2024-05-12T00:00:00.222Z","_stream":"{}","_msg":"m2"}


# example 2

# works as expected here
curl 'http://localhost:9428/select/logsql/query' --data-urlencode \
    'query=_time:[0,2024-05-12T00:00:00.555+00:00]  '
{"_time":"2024-05-12T00:00:00.111Z","_stream":"{}","_msg":"m1"}
{"_time":"2024-05-12T00:00:00.222Z","_stream":"{}","_msg":"m2"}
{"_time":"2024-05-12T00:00:00.333Z","_stream":"{}","_msg":"m3"}
{"_time":"2024-05-12T00:00:00.444Z","_stream":"{}","_msg":"m4"}
{"_time":"2024-05-12T00:00:00.555Z","_stream":"{}","_msg":"m5"}

# requested range has exclusive upper bound, but 
# boundary point m5 gets included into result
curl 'http://localhost:9428/select/logsql/query' --data-urlencode \
    'query=_time:[0,2024-05-12T00:00:00.555+00:00)  '
{"_time":"2024-05-12T00:00:00.111Z","_stream":"{}","_msg":"m1"}
{"_time":"2024-05-12T00:00:00.222Z","_stream":"{}","_msg":"m2"}
{"_time":"2024-05-12T00:00:00.333Z","_stream":"{}","_msg":"m3"}
{"_time":"2024-05-12T00:00:00.444Z","_stream":"{}","_msg":"m4"}
{"_time":"2024-05-12T00:00:00.555Z","_stream":"{}","_msg":"m5"}

```

Probably related to float, nonetheless this doesn't look right.

### Version

`victoria-logs-20240515-025846-tags-v0.7.0-victorialogs-0-g6fdba8599d`

**Comment by Amper** [comment]
Hey @eduardbareev.
This issue should be fixed after [this PR](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6295).

**Comment by valyala** [comment]
The issue has been fixed in [VictoriaLogs v0.9.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v0.9.0-victorialogs).
